Merge css-docs > tech-docs (including rework of Warp examples etc)

magnuh opened the pull request.

Commit:
docs: Merging css-docs > tech-docs

Commit:
docs: merge conflict

Commit:
fix: merge conflicts

Commit:
docs: add css-docs components and background-color docs

magnuh committed:
fix: messing around with uno configs (#194)

YOLO!

magnuh committed:
messing more with uno configs

Commit:
docs: add all css-docs and images

magnuh committed:
fixing sidebar link styling

magnuh committed:
Merge remote-tracking branch 'origin/classdocs' into classdocs

magnuh committed:
remove obsolete unsupported notes

Commit:
fix: background-attachment examples

imprashast committed:
Some cleanup

Commit:
fix: bac to tex

magnuh committed:
merge from master

Commit:
Add missing intropages to the tech docs

magnuh committed:
cleaned up Container a bit

magnuh committed:
Update docs/.vitepress/config.js

Co-authored-by: Balbina K <41303231+BalbinaK@users.noreply.github.com>

Commit:
Added a very simple explanation on how unocss works and vs tailwind

Skadefryd committed:
Update docs/.vitepress/config.js

Fix: typo

Co-authored-by: Balbina K <41303231+BalbinaK@users.noreply.github.com>

magnuh committed:
Merge branch 'master' of https://github.com/warp-ds/tech-docs into classdocs

magnuh committed:
Started rework of all Warp examples

magnuh committed:
Rework of all the Typography examples

magnuh committed:
Rework of the Iconography examples

magnuh committed:
Rework of the Background examples

magnuh committed:
Merge from master

magnuh committed:
Merge from classdocs

magnuh committed:
Rework of Padding examples

magnuh committed:
Rework of the rest of the Spacing examples

magnuh committed:
Let's just pre-create all the presetDocs color classes for bg and border...

magnuh committed:
Rework of flex examples

magnuh committed:
Rework of grid examples

magnuh committed:
Rework of remaining flex and grid examples

magnuh committed:
Rework of border, divide and outline examples

magnuh committed:
Rework of Effects and Filters examples

magnuh committed:
Rework of Table examples

magnuh committed:
Rework of Transitions and Animations examples

magnuh committed:
Rework of Transforms examples

magnuh committed:
Rework of Interactivity examples

magnuh committed:
Minor tweaks for misc component examples

magnuh committed:
A few more example tweaks...

magnuh committed:
Removed old Container component (in favor of example-container)

magnuh committed:
Removed old Box component, used mainly for striped backgrounds in examples (in favor of ex-bg--*)

magnuh committed:
Tweaking paragraph order

magnuh committed:
Removed all theme-switchers from the docs pages and added a single one to the header. Chosen theme is stored in localStorage.

magnuh committed:
Make sure chosen theme is updated in shadow-dom when navigating between pages.

magnuh committed:
Remove references / links to css-docs

magnuh committed:
Relative paths in bg-urls will hopefully solve problem with different base paths in dev and prod

magnuh committed:
Merge from master

magnuh committed:
Merge branch 'classdocs' of https://github.com/warp-ds/tech-docs into cssdocs-including-rework-of-warp-examples

magnuh committed:
Remove "under construction sign"

magnuh committed:
Fixes for the Divide Color examples

magnuh committed:
Fixed broken images from Unsplash image service by changing to local image assets

magnuh committed:
Deprecate notification variant of Badge

magnuh committed:
Update info in "Collaborate with us" section

magnuh committed:
Add links to Slack channels where missing

magnuh committed:
Remove duplication of "Getting started" (and minor tweaks)

magnuh committed:
Migration docs tweaks

magnuh committed:
Dark mode adjustments to tip boxes

magnuh committed:
Class variants tweaks

magnuh committed:
Class states tweaks

magnuh committed:
Tweaks to the component status display

magnuh committed:
Tweaks to the icons docs

magnuh committed:
Tweaks to the Card docs

magnuh committed:
Updated breadcrumbs examples

magnuh committed:
Attention examples tweaks... and misc

magnuh committed:
cleaning

magnuh committed:
Tweak and improve icons documentation

magnuh committed:
Rename/move "Classes" folders/menu item to "CSS"

magnuh committed:
merge from master

magnuh committed:
Merge branch 'classdocs' of https://github.com/warp-ds/tech-docs into cssdocs-including-rework-of-warp-examples

magnuh committed:
Rename Classes menu item and update dependencies

magnuh committed:
config cleanup

BalbinaK commented on docs/getting-started/developers/index.md:
```diff
@@ -6,11 +6,15 @@
+Only the used utilities are shipped to your app, ensuring speed and efficiency in any project size.
 
-If you are migrating from Fabric to Warp, please visit the [Migration page](/migration/developers/).
+If you are migrating from Fabric to Warp, please visit the [Migration page](/migration/developers/.
```
```suggestion
If you are migrating from Fabric to Warp, please visit the [Migration page](/migration/developers/).
```

BalbinaK commented on docs/.vitepress/config.js:
```diff
@@ -31,18 +112,25 @@ export default defineConfig({
+          'ex-inner-box': 'p-24 rounded pd-shadow-xl mx-auto pd-bg-white pd-text-slate-500 max-w-[300]',
+          'ex-pic-no': 'absolute top-10 left-10 h-32 w-32 text-center pd-bg-white rounded-full pd-text-slate-800 leading-[32]',
+        }],
+        safelist: [...classes, ...supported, ...docsClasses],
```
Would it make sense to rename those to something like `...componentClasses, ...supportedClasses` (or `...cssClasses`?). Just so it's easier to associate with respective part of the docs.

BalbinaK commented on docs/.vitepress/theme/index.js:
```diff
@@ -7,19 +8,34 @@ import ComponentQuestions from '../../components/ComponentQuestions.md';
-    app.component('DoDont', DoDont);
+  Layout() {
+    return h(DefaultTheme.Layout, null, {
+      'nav-bar-content-before': () => h(ThemeSwitcher)
```
Would it make sense to have the theme switcher appear after the search field? I have a feeling it's not that relevant to most users what theme is applied. 
Also, when the select element is in focus, it gets a red outline, while other interactive elements on that page have blue outline color.
Finally, a small thing, but could we add some horizontal padding to the content div so the switcher is a bit further to the right.
![Screenshot 2024-06-10 at 14 12 18](https://github.com/warp-ds/tech-docs/assets/41303231/f3c7f6cc-4199-4df9-a880-859c1e6e85ba)

magnuh replied:
Pushed tweaks to the theme switcher: 
- Moved it to the right of the search field
- Added it to the mobile screen menu
- Minor layout fixes etc

BalbinaK commented on docs/css/aspect-ratio.md:
```diff
@@ -0,0 +1,72 @@
+
+## Quick reference
+
+::: warning Warning
+  Known problem: the shortcuts `aspect-ratio`, `aspect-square`, `aspect-video` are not working as intended.
+:::
```
Just checking, is this still valid @Skadefryd?

Skadefryd replied:
Still applies, we are still on the margin-trick for this , and those are part of the css native aspect-ratio .  (Does not interchange because the margin trick way of doing this needs to be set on a parent while aspect-ratio as  css is set directly on the thing.

BalbinaK commented on docs/css/font-size.md:
```diff
@@ -0,0 +1,84 @@
+
+### Setting the headings
+
+Control the heading sizes using the `h1` - `h5` utilities. These classes aim to contain all the styling needed for each heading-level pr. brand.
```
I think we should advise using (and use) the same classes Figma is exposing, which is `t1-t5`. h1-h5 is something that came with Fabric and we made shortcuts in Warp that would map those to t1-t5, which generate the same style.

BalbinaK commented on docs/css/background-color.md:
```diff
@@ -0,0 +1,103 @@
+This works for both semantic and arbitrary classes with any primitive `--w-{color}` or semantic `--w-s-color-{name}` css variable.
+
+::: warning Warning
+Please be advised that by using primitive variables you will lose theming possibilities, therefor we highly recommend only using semantic classes and tokens.
```
```suggestion
Please be advised that by using primitive variables you will lose theming possibilities, therefore we highly recommend only using semantic classes and tokens.
```

BalbinaK commented on docs/css/scroll-snap-stop.md:
```diff
@@ -0,0 +1,18 @@
+
+## Quick reference
+
+<qr-table />
```
Some unsupported utility classes are formatted with a line through text, and some aren't, but in the [css-docs](https://warp-ds.github.io/css-docs/scroll-snap-stop) all have such formatting.
![Screenshot 2024-06-10 at 14 38 35](https://github.com/warp-ds/tech-docs/assets/41303231/1aa31f31-1d52-4437-83f0-4413dbbd4136)

magnuh replied:
Fixed.

BalbinaK commented on docs/components/icons/Example.vue:
```diff
@@ -3,35 +3,23 @@ import decamelize from "decamelize";
+// When an existing icon is deprecated, add { old: "IconOldName16", new: "IconNewName16" } to the deprecatedIcons list.
 // This will display a deprecation message with a suggested replacement.
-const deprecatedIcons = []
+const deprecatedIcons = [{old:'IconAds16', new:'IconAirCon16'},{old:'IconHonkDark42'}];
```
Is IconAds16 and IconHonkDark42 deprecated, or was it a display of how we could present deprecated icons (which I like btw)?
![Screenshot 2024-06-10 at 14 47 14](https://github.com/warp-ds/tech-docs/assets/41303231/ccc5abb0-1f26-44ac-bad6-5cb192f7c767)

magnuh replied:
Ah. Crap. That's just for testing... 😅 I'll remove it. 👍

BalbinaK commented on docs/components/attention/Example.vue:
```diff
@@ -85,7 +84,7 @@ const popoverIconTargetShowing = ref(false)
     </div>
     <div>
-      <h4>Popover with icon as target element</h4>
+      <h3 class="t4">Popover with icon as target element</h3>
       <w-button
         :aria-expanded="popoverIconTargetShowing"
         aria-controls="popover-icon-target-example"
```
Not sure what has caused this, but in prod this icon is not clipped on the right side, like it is on this branch
![Screenshot 2024-06-10 at 14 51 11](https://github.com/warp-ds/tech-docs/assets/41303231/2a9073da-5575-4002-9b8c-c0c5ebbce47e)

magnuh replied:
? 🤷 
![image](https://github.com/warp-ds/tech-docs/assets/7531505/ae707cb2-3cce-4f79-8d67-9b352015c680)

felicia-haggqvist replied:
I don't get it clipped either. 🤔  Tested in Chrome, Safari and Firefox: 
<img width="57" alt="Skärmavbild 2024-06-11 kl  16 03 16" src="https://github.com/warp-ds/tech-docs/assets/144792260/bc402ee9-2366-4181-ba77-b27e66926234">

BalbinaK replied:
Whoa, I restarted Chrome and it's correct now 😄

BalbinaK reviewed (commented):
Nice! Below a few minor comments from my side.

felicia-haggqvist commented on docs/css/border-color.md:
```diff
@@ -0,0 +1,129 @@
+This works for both semantic and arbitrary classes with any primitive `--w-{color}` or semantic `--w-s-color-{name}` css variable.
+
+::: warning Warning
+Please be advised that by using primitive variables you will lose theming possibilities, therefor we highly recommend only using semantic classes and tokens.
```
```suggestion
Please be advised that by using primitive variables you will lose theming possibilities, therefore we highly recommend only using semantic classes and tokens.
```

felicia-haggqvist commented on docs/css/divide-color.md:
```diff
@@ -0,0 +1,97 @@
+This works for both semantic and arbitrary classes with any primitive `--w-{color}` or semantic `--w-s-color-{name}` css variable.
+
+::: warning Warning
+Please be advised that by using primitive variables you will lose theming possibilities, therefor we highly recommend only using semantic classes and tokens.
```
```suggestion
Please be advised that by using primitive variables you will lose theming possibilities, therefore we highly recommend only using semantic classes and tokens.
```

felicia-haggqvist commented on docs/css/text-color.md:
```diff
@@ -0,0 +1,117 @@
+This works for both semantic and arbitrary classes with any primitive `--w-{color}` or semantic `--w-s-color-{name}` css variable.
+
+::: warning Warning
+Please be advised that by using primitive variables you will lose theming possibilities, therefor we highly recommend only using semantic classes and tokens.
```
```suggestion
Please be advised that by using primitive variables you will lose theming possibilities, therefore we highly recommend only using semantic classes and tokens.
```

felicia-haggqvist commented on docs/css/icon-color.md:
```diff
@@ -0,0 +1,70 @@
+This works for the semantic icon classes, and you can also use the arbitrary text classes with any primitive `--w-{color}` or semantic `--w-s-color-{name}` css variable.
+
+::: warning Warning
+Please be advised that by using primitive variables you will lose theming possibilities, therefor we highly recommend only using semantic classes and tokens.
```
```suggestion
Please be advised that by using primitive variables you will lose theming possibilities, therefore we highly recommend only using semantic classes and tokens.
```

felicia-haggqvist commented on docs/css/outline-color.md:
```diff
@@ -0,0 +1,92 @@
+This works for both semantic and arbitrary classes with any primitive `--w-{color}` or semantic `--w-s-color-{name}` css variable.
+
+::: warning Warning
+Please be advised that by using primitive variables you will lose theming possibilities, therefor we highly recommend only using semantic classes and tokens.
```
```suggestion
Please be advised that by using primitive variables you will lose theming possibilities, therefore we highly recommend only using semantic classes and tokens.
```

magnuh committed:
Fixed dark mode for visibility example

magnuh committed:
Fixed link typo

magnuh committed:
Fix example background color in color tables for dark mode

magnuh committed:
Better names for safelist classes

magnuh committed:
Tweaks to the theme switcher:
- Move it to the right of the search field
- Add it to the mobile screen menu
- Minor layout fixes

magnuh committed:
Changed heading docs to use t1-t5 utility classes instead of h1-h5

magnuh committed:
Therefore

magnuh committed:
Remove testing purpose icon deprecation

magnuh committed:
Add line-through for unsupported attributes/classes in quick reference table

felicia-haggqvist reviewed (approved):
LGTM! 🎉  Really nice work! 👏

Skadefryd commented on docs/css/background-position.md:
```diff
@@ -0,0 +1,90 @@
+    </div>
+    <div class="flex flex-col justify-self-center place-items-center">
+      <div class="pb-8 pd-font-mono">bg-left-bottom</div>
+      <div class="h-128 w-128 rounded-8 bg-bottom-left bg-auto bg-[url(./50s-scientists.jpg)]" style="background-size:300%"></div>
```
<img width="759" alt="image" src="https://github.com/warp-ds/tech-docs/assets/462825/5369283b-bc7d-4687-bc0b-6a172fc2287e">
2fc2287e">

I dont see why but some of the classes like the bg-bottom-left and bg-bottom-right are not getting any styles generated , maybe it turned arround, is it bg-bottom-left, or bg-left-bottom ?

magnuh replied:
The code example was correct, but the visual example was mixed up. 😅  Fixed now. Thanks. 👍

Skadefryd reviewed (changes requested):
Stellar!

Just a small i think wrong used classes in the example for background position.   .bg-bottom-left works on drive, but example uses .bg-left-bottom

magnuh committed:
Fixed background position examples

Skadefryd reviewed (approved):
LGTM
